### PR TITLE
refactor(ssl.proto): extracted from `v1` package to `ssl` package

### PR DESCRIPTION
**Pull request by gunnar-solo**
# Description
TL;DR:
make this change:
`ssl.proto`
```proto
-option go_package = "github.com/solo-io/gloo/projects/gloo/pkg/api/v1";
+option go_package = "github.com/solo-io/gloo/projects/gloo/pkg/api/v1/ssl";
```

then fix resulting build errors

# Context
~This is meant to support https://github.com/solo-io/gloo/issues/7693 by preventing a circular dependency caused by adding an `ssl_config` field to `dynamic_forward_proxy.proto`.  It both~
~a) resolves said circular dependency via refactor~
~b) adds the field, to prove it _is_ resolved~
~a follow-up PR will actually _consume_ said new field~
context was asked to be removed, so it was

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: changelog/v1.14.0-beta10/7693-refactor-ssl.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+changelog:
+- type: NON_USER_FACING
+  descrption: Extract the generated files from `ssl.proto` into their own go package.  This is meant to support a follow-up PR by resolving a circular dependency
```
Comment (nfuden) [inline changelog/v1.14.0-beta10/7693-refactor-ssl.yaml:3]:
```suggestion
  description: Extract the generated files from `ssl.proto` into their own go package.  This is meant to support a follow-up PR by resolving a circular dependency where ssl can be used in more places.
```

File: projects/gloo/api/v1/ssl.proto [hunk 2]
```diff
@@ -1,6 +1,6 @@
 syntax = "proto3";
 package gloo.solo.io;
-option go_package = "github.com/solo-io/gloo/projects/gloo/pkg/api/v1";
```
Comment (elcasteel) [inline projects/gloo/api/v1/ssl.proto:3]: I'm not super well versed in conventions here, but should we also move this to a new directory?

**Comment by elcasteel** [comment]
Have you attempted the solo-projects side of this change? Maybe worth doing a draft of that so that we don't accidentally prevent ourselves from being able to bump to a new gloo version if it turns out to require more than a few hours of work.

File: changelog/v1.14.0-beta10/7693-refactor-ssl.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+changelog:
+- type: NON_USER_FACING
+  descrption: description: Extract the generated files from `ssl.proto` into their own go package.  This is meant to support a follow-up PR by resolving a circular dependency where ssl can be used in more places.
```
Comment (jbohanon) [inline changelog/v1.14.0-beta10/7693-refactor-ssl.yaml:3]: looks like LCB is displeased with this yaml

File: changelog/v1.14.0-beta10/7693-refactor-ssl.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+changelog:
+- type: NON_USER_FACING
```
Comment (nfuden) [inline changelog/v1.14.0-beta10/7693-refactor-ssl.yaml:2]: Is this breaking for people using gloo as a library for their controllers?

Comment (jenshu) [reply changelog/v1.14.0-beta10/7693-refactor-ssl.yaml:2]: yes. i wish solo-apis had its own changelogs

File: install/helm/gloo/crds/gateway.solo.io_v1_Gateway.yaml [hunk 5]
```diff
@@ -340,6 +340,76 @@ spec:
                             type: object
                           saveUpstreamAddress:
                             type: boolean
+                          sslConfig:
```
Comment (jenshu) [inline install/helm/gloo/crds/gateway.solo.io_v1_Gateway.yaml:343]: do we know what's causing the CRD updates?
moving the import to another pkg shouldn't have an effect on the CRDs, right?

File: projects/gloo/api/v1/options/dynamic_forward_proxy/dynamic_forward_proxy.proto [hunk 6]
```diff
@@ -27,6 +28,9 @@ message FilterConfig {
   // `envoy.stream.upstream_address` (See
   // :repo:`upstream_address.h<source/common/stream_info/upstream_address.h>`).
   bool save_upstream_address = 2;
+
+  // If provided, the autogenerated dynamic forward proxy cluster will use TLS when connecting to upstream hosts and verify the certificate chain
+  .gloo.solo.io.UpstreamSslConfig ssl_config = 3;
```
Comment (jenshu) [inline projects/gloo/api/v1/options/dynamic_forward_proxy/dynamic_forward_proxy.proto:33]: oh i see. should this be removed from this PR and added to the follow-up instead?

**Review by jenshu — CHANGES_REQUESTED** [review]
LGTM aside from the addition in dfp proto which should be a separate PR

**Review by jenshu — APPROVED** [review]

**Review by nfuden — APPROVED** [review]
